test/integ/cli/test_cli_asset.py

```diff
@@ -0,0 +1,126 @@
+            f.write("test file content")
+
+        runner = CliRunner()
+        runner.invoke(main, ["asset", "snapshot", "--root-dir", root_dir])
+
+        # Check manifest file details to match correct content
+        # since manifest file name is hashed depending on source location, we have to list out manifest
+        manifest_folder_path = os.path.join(root_dir, f"{os.path.basename(root_dir)}_manifests")
+        manifest_file_name = os.listdir(manifest_folder_path)[0]
+        manifest_file_path = os.path.join(manifest_folder_path, manifest_file_name)
+
+        with open(manifest_file_path, "r") as f:
+            manifest_data = json.load(f)
+            assert manifest_data["paths"][0]["path"] == "file.txt"
+            assert (
+                manifest_data["paths"][0]["hash"] == "0741993e50c8bc250cefba3959c81eb8"
+            )  # hashed with xxh128
+
+    def test_raises_on_invalid_root_dir(self, temp_dir):
+        """
+        Snapshot with invalid root directory, raises exception
+        """
+        invalid_root_dir = os.path.join(temp_dir, "invalid_root_dir")
+        runner = CliRunner()
+        result = runner.invoke(main, ["asset", "snapshot", "--root-dir", invalid_root_dir])
+        assert result.exit_code == 1
+
+    def test_pass_valid_root_dir_recursive(self, temp_dir):
+        """
+        Snapshot with valid root directory and recursively
+        """
+        root_dir = os.path.join(temp_dir, "root_dir")
+
+        # Create a file in the root directory
+        root_file_path = os.path.join(root_dir, "root_file.txt")
+        with open(root_file_path, "w") as f:
+            f.write("root file content")
+
+        # Create a file in the subdirectory
+        subdir_file_path = os.path.join(root_dir, "subdir1", "subdir2", "subdir_file.txt")
+        os.makedirs(os.path.dirname(subdir_file_path), exist_ok=True)
+        with open(subdir_file_path, "w") as f:
+            f.write("subdir file content")
+
+        runner = CliRunner()
+        runner.invoke(main, ["asset", "snapshot", "--root-dir", root_dir, "--recursive"])
+
+        # Check manifest file details to match correct content
+        # since manifest file name is hashed depending on source location, we have to list out manifest
+        manifest_folder_path = os.path.join(root_dir, f"{os.path.basename(root_dir)}_manifests")
+        root_manifest_file_name = [file for file in os.listdir(manifest_folder_path)][0]
+        root_manifest_file_path = os.path.join(manifest_folder_path, root_manifest_file_name)
+        with open(root_manifest_file_path, "r") as f:
+            manifest_data = json.load(f)
+            assert manifest_data["paths"][0]["path"] == "root_file.txt"
+            assert (
+                manifest_data["paths"][0]["hash"] == "a5fc4a07191e2c90364319d2fd503cc1"
+            )  # hashed with xxh128
+            assert manifest_data["paths"][1]["path"] == posixpath.join(
+                "subdir1", "subdir2", "subdir_file.txt"
```

Comment:
In general for tests, the test input can be defined as a variable and pass to the call and assertion. Take a look at how it's done in other test cli implementation.